src/runtime_src/core/common/api/aie/xrt_graph.cpp

```diff
@@ -189,27 +233,26 @@ wait_gmio(xrtDeviceHandle dhdl, const char *gmio_name)
   device->wait_gmio(gmio_name);
 }
 
-static int
-start_profiling(xrtDeviceHandle dhdl, int option, const char *port1_name, const char *port2_name, uint32_t value)
-{
-  auto device = xrt_core::device_int::get_core_device(dhdl);
-  return device->start_profiling(option, port1_name, port2_name, value);
-}
+// C-API Event handles are inserted to this map.
+static std::map<int, std::shared_ptr<xrt::aie::event_impl>> event_cache;
 
-static uint64_t
-read_profiling(xrtDeviceHandle dhdl, int phdl)
+static std::shared_ptr<xrt::aie::event_impl>
+start_profiling(xrtDeviceHandle dhdl, int option, const char *port1_name, const char *port2_name, uint32_t value)
```

Comment:
these can be part of event_impl class functions once you make all other changes.